azurerm/config.go

```diff
@@ -264,9 +265,30 @@ type ArmClient struct {
 	policyDefinitionsClient policy.DefinitionsClient
 }
 
+var (
+	msClientRequestIDOnce sync.Once
+	msClientRequestID     string
+)
+
+// clientRequestID generates a UUID to pass through `x-ms-client-request-id` header.
+func clientRequestID() string {
+	msClientRequestIDOnce.Do(func() {
+		var err error
+		msClientRequestID, err = uuid.GenerateUUID()
+
+		if err != nil {
+			log.Printf("[WARN] Fail to generate uuid for msClientRequestID: %+v", err)
+		}
```

Comment:
Should we return nil here since we hit an error and not let it fall through?

Reply:
> Should we return nil here since we hit an error and not let it fall through?

This client request id is not blocking step to call ARM API, so we don't need to fail fast.